### PR TITLE
e2e: serial: handle missing NodeGroupConfig in spec

### PR DESCRIPTION
**Pull request by fromanirh**
The NUMAResourceOperator spec may miss the NodeGroupConfig spec, and the tests should handle this state gracefully.

Additionally, wrap the main testing logic in a Eventually block, because the status can take nonzero time to be correctly populated, and this should not make the test fail.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

File: test/e2e/serial/tests/configuration.go [hunk 1]
```diff
@@ -538,19 +539,25 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
...
+
+				// normalize config to handle unspecified defaults
+				specConf := nropv1alpha1.DefaultNodeGroupConfig()
+				if nroOperObj.Spec.NodeGroups[0].Config != nil {
+					specConf = merge.NodeGroupConfig(specConf, *nroOperObj.Spec.NodeGroups[0].Config)
+				}
 
+				statusConf := nroOperObj.Status.MachineConfigPools[0].Config // shortcut
+				g.Expect(statusConf).ToNot(BeNil())
+				g.Expect(statusConf).To(Equal(specConf))
```
Comment (fromanirh) [inline test/e2e/serial/tests/configuration.go:559]: I should mention explicitely in a comment that the status must report the default config even if the spec is missing.

File: controllers/numaresourcesoperator_controller.go [hunk 2]
```diff
@@ -227,7 +227,7 @@ func (r *NUMAResourcesOperatorReconciler) reconcileResource(ctx context.Context,
 	}
 	r.Recorder.Eventf(instance, corev1.EventTypeNormal, "SuccessfulRTECreate", "Created Resource-Topology-Exporter DaemonSets")
 
-	instance.Status.MachineConfigPools = syncMachineConfigPoolNodeGroupConfigStatuses(instance.Name, instance.Status.MachineConfigPools, trees)
+	instance.Status.MachineConfigPools = syncMachinetConfigPoolNodeGroupConfigStatuses(instance.Name, instance.Status.MachineConfigPools, trees)
```
Comment (shajmakh) [inline controllers/numaresourcesoperator_controller.go:230]: was this modified by mistake?  I don't see a matching function

Comment (fromanirh) [reply controllers/numaresourcesoperator_controller.go:230]: yes, a typo sneaked in. Thanks!

File: test/e2e/serial/tests/configuration.go [hunk 3]
```diff
@@ -538,19 +539,29 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
...
+				}
 
+				// the status must be always populated by the operator.
+				// If the user-provided spec is missing, the status must reflect the compiled-in defaults.
+				// This is wrapped in a Eventually because even in functional, well-behaving clusters,
+				// the operator may take nonzero time to populate the status, and this is still fine.
+				statusConf := nroOperObj.Status.MachineConfigPools[0].Config // shortcut
+				g.Expect(statusConf).ToNot(BeNil())
+				g.Expect(statusConf).To(Equal(specConf))
+			}).WithTimeout(5*time.Minute).WithPolling(10*time.Second).Should(Succeed(), "failed to check the NodeGroupConfig status for %q", nroKey.String())
```
Comment (shajmakh) [inline test/e2e/serial/tests/configuration.go:564]: new for me to see Eventually with Gomega, that's nice and useful!

**Comment by shajmakh** [comment]
Thanks! 
/lgtm

File: test/e2e/serial/tests/configuration.go (hunk 3 above)
Comment (fromanirh) [reply test/e2e/serial/tests/configuration.go:564]: Learned myself not so long ago :) the [godocs fo gomega](https://pkg.go.dev/github.com/onsi/gomega#Eventually) have been a very nice read. There's also more to explore here!

Comment (shajmakh) [reply test/e2e/serial/tests/configuration.go:564]: Thanks for the reference, I'll definitely check it!

**Pull request merged**